internal/testrunner/runners/system/servicedeployer/custom_agent.go

```diff
@@ -110,12 +157,6 @@ func (d *CustomAgentDeployer) SetUp(inCtxt ServiceContext) (DeployedService, err
 		return nil, errors.Wrap(err, "could not boot up service using Docker Compose")
 	}
 
-	// Connect service network with stack network (for the purpose of metrics collection)
-	err = docker.ConnectToNetwork(p.ContainerName(serviceName), stack.Network())
-	if err != nil {
-		return nil, errors.Wrapf(err, "can't attach service container to the stack network")
-	}
```

Comment:
I moved this to the docker-compose file as I've seen [this call fail](https://github.com/elastic/elastic-agent/blob/main/internal/pkg/agent/cmd/container.go#L494) in agent containers. I assumed it is caused by the period where the agent is started but not attached yet, and it is, but the retry mechanism eventually succeeds so this is not necessary